### PR TITLE
[GCAL/MSCAL] Create event UX improvements

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary
- Added "Location" and "Link to channel" optional field text
- Time pickers now have defaults depending on the selected date
- Custom icon in "Create event" menu item
  Screenshot 2023-08-25 at 13 43 56
- Added custom icon to date selector:
  Screenshot 2023-08-25 at 13 56 35

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53998

### CODE REVIEW

File: webapp/src/components/modals/create_event_form.css [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.create-event-form input[type="date"]::-webkit-calendar-picker-indicator {
```
Comment (mickmister) [inline webapp/src/components/modals/create_event_form.css:1]: Can we use a more specific class name, like `mscalendar-create-event-form`? This way it's definitely scoped to this plugin.

File: webapp/src/custom.d.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+declare module '*.svg' {
```
Comment (mickmister) [inline webapp/src/custom.d.ts:1]: :+1: A pain that this is required though

File: webapp/tsconfig.json [hunk 3]
```diff
@@ -25,6 +27,9 @@
     "experimentalDecorators": true,
     "jsx": "react"
   },
+  "files": [
+    "src/custom.d.ts"
+  ],
```
Comment (mickmister) [inline webapp/tsconfig.json:32]: I don't think I've ever used `files` like this for `.d.ts` files. Maybe we can move `custom.d.ts` into `types/svg.d.ts`, and then set tsconfig's `typeRoots` to `./types`. I think that's the proper way to do this.

But if it ain't broke, don't fix it? 🤷

File: webapp/src/index.tsx [hunk 4]
```diff
@@ -27,7 +30,7 @@ export default class Plugin {
             await store.dispatch(getProviderConfiguration());
 
             registry.registerChannelHeaderMenuAction(
-                <span><i className='icon fa fa-calendar-plus-o'/>{'Create calendar event'}</span>,
+                <span><Icon image={CalendarOutlineIcon}/>{'Create calendar event'}</span>,
```
Comment (mickmister) [inline webapp/src/index.tsx:33]: This line is getting kind of long

```suggestion
                (
                    <span>
                        <Icon image={CalendarOutlineIcon}/>
                        {'Create calendar event'}
                    </span>
                ),
```

0/5 on the parens that I added there, but definitely think it should be multiple lines

File: webapp/src/components/icon.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            <i
+                className='icon fa'
+                style={{
+                    width: '18px',
+                    height: '18px',
+                    backgroundImage: `url(${this.props.image})`,
+                    backgroundPosition: '50% 50%',
+                    backgroundRepeat: 'no-repeat',
+                    verticalAlign: 'middle',
+                }}
```
Comment (mickmister) [inline webapp/src/components/icon.tsx:27]: Is this taken from somewhere else? It seems very specific (size etc.) for a generic-sounding `Icon` component. It seems fine though.

File: webapp/src/components/modals/create_event_form.css (hunk 1 above)
Comment (mickmister) [reply webapp/src/components/modals/create_event_form.css:1]: Also we generally prefer `scss` over `css` files, which allow you to do things like nesting:

```suggestion
.create-event-form {
    input[type="date"]::-webkit-calendar-picker-indicator {
        background: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath d='M12 12H17V17H12V12M19 3H18V1H16V3H8V1H6V3H5C3.9 3 3 3.9 3 5V19C3 20.1 3.9 21 5 21H19C20.1 21 21 20.1 21 19V5C21 3.9 20.1 3 19 3M19 5V7H5V5H19M5 19V9H19V19H5Z' /%3E%3C/svg%3E") no-repeat;
        display: block;
        height: 18px;
        width: 18px;
    }
}
```

**Review by mickmister — APPROVED** [review]
LGTM, a few non-blocking suggestions

File: webapp/src/custom.d.ts (hunk 2 above)
Comment (fmartingr) [reply webapp/src/custom.d.ts:1]: You tell me. 😅

File: webapp/src/components/icon.tsx (hunk 5 above)
Comment (fmartingr) [reply webapp/src/components/icon.tsx:27]: I just played with the styles until it seemed right.

File: webapp/tsconfig.json (hunk 3 above)
Comment (fmartingr) [reply webapp/tsconfig.json:32]: Let's do it the proper way.

File: webapp/src/components/modals/create_event_form.css (hunk 1 above)
Comment (mickmister) [reply webapp/src/components/modals/create_event_form.css]: Thoughts about using the class name `mscalendar-create-event-form` instead? We usually try to scope css classes to a given plugin whenever possible

**Review by mickmister — APPROVED** [review]

File: webapp/src/components/modals/create_event_form.css (hunk 1 above)
Comment (fmartingr) [reply webapp/src/components/modals/create_event_form.css]: Oh yeah I got lost in thought about using `mscalendar` as class name and I forgot to put it while making other changes.

**Pull request merged**